apluslms_roman/backends/docker.py

```diff
@@ -99,20 +103,41 @@ def _run_opts(self, task, step):
 
     def prepare(self, task, observer):
         client = self._client
+        images = self._cache.images
+        day = timedelta(days=1)
         for step in task.steps:
             observer.step_preflight(step)
             image, tag = step.img.split(':', 1)
+
+            last_updated = images.get(step.img, None)
+            last_updated = (day if not last_updated
+                else datetime.now() - last_updated)
+
             try:
-                img = client.images.get(step.img)
+                client.images.get(step.img)
+                img_found = True
             except docker.errors.ImageNotFound:
+                img_found = False
+
+            if not img_found or last_updated >= day:
                 observer.step_running(step)
-                observer.manager_msg(step, "Downloading image {}".format(step.img))
+                if img_found:
+                    observer.manager_msg(step,
+                        "Checking for updates for {}".format(step.img))
```

Comment:
Is this message accurate enough...? It kind of sounds like it'll only download the image again if there are updates but 'checking for updates' just means it downloads the image again.

Reply:
Kind of would be good to first say `checking for updates` and then `downloading`, but that feels a lot of text (I think I commented about that before). As the process is actually `trying to get update`, I think we should indicate that we do both (check and download).

Reply:
hmmm `Downloading in case of updates`? does that sound good?

Reply:
That or `Checking for updates and downloading if any`.

Reply:
Is there a way to actually do that? Because currently it just downloads the image without really doing any checks.

Reply:
No. Check and download is more human way to explain synchronisation.